odc/geo/roi.py

```diff
@@ -77,7 +77,7 @@ def chunks(self) -> Chunks2d: ...
 
     def locate(self, pix: SomeIndex2d) -> Tuple[int, int]: ...
 
-    def __dask_tokenize__(self): ...
+    def __dask_tokenize__(self) -> None: ...
```

Comment:
pretty sure this function is supposed to return something

Reply:
https://github.com/dask/dask/blob/3a810721c0d6bba760cbf472f9ae907e809de92c/dask/typing.py#L215-L218

```python
    @abc.abstractmethod
    def __dask_tokenize__(self) -> Hashable:
        """Value that must fully represent the object."""
        raise NotImplementedError("Inheriting class must implement this method.")
```

Reply:
I've updated the return type, but I would have expected an `@override` on the method so I think there's something more that should be fixed in this area. That is a separate issue though, I have no idea how these things are supposed to be stringed together.

Reply:
Does `@override` even makes sense on a `Protocol` type? It's really just formalizing duck-typing expected by your code.

Reply:
I don't know anything about dask, but the `DaskCollection` you linked has bothered to make the method an abstract method through abc, so it's pretty clear that they intend subclasses to override and implement that method along with the other abstract methods.